### PR TITLE
chore: explicit DescribeCluster panic message on ZS enablement

### LINKED ISSUE #9151: ValidateZonalShiftEnablement panics with nil pointer dereference when eks:DescribeCluster fails
### Description

**Observed Behavior**:
Karpenter panics when DescribeCluster is not allowed and zonal shift is enabled
```
  panic: runtime error: invalid memory address or nil pointer dereference
  [signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x2c6f6f8]

  goroutine 1 [running]:
  github.com/aws/karpenter-provider-aws/pkg/operator.ValidateZonalShiftEnablement
      github.com/aws/karpenter-provider-aws/pkg/operator/operator.go:360 +0xd8

  Root cause in [operator.go:360](https://github.com/aws/karpenter-provider-aws/blob/v1.12.0/pkg/operator/operator.go#L360):

  outputDC, _ := eksAPI.DescribeCluster(ctx, &inputDC)
  clusterArn := outputDC.Cluster.Arn  // nil dereference if DescribeCluster failed
```

**Expected Behavior**:
Karpenter should probably report errors and use default behavior, or should panic with a message indicating that the IAM setup is incorrect.

**Reproduction Steps** (Please include YAML):

**Versions**:
- Chart Version: v1.12.0
- Kubernetes Version (`kubectl version`): 1.35

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### PR DESCRIPTION
**Pull request by ryan-mist**
Fixes #9151

**Description**
* catches errors when describe cluster during zonal shift enablement

**How was this change tested?**
* make presubmit

**Does this change impact docs?**
- [ ] Yes, PR includes docs updates 
- [ ] Yes, issue opened: # 
- [x] No

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: kwok/operator/operator.go [hunk 1]
```diff
@@ -355,7 +355,10 @@ func SetupIndexers(ctx context.Context, mgr manager.Manager) {
 
 func ValidateZonalShiftEnablement(ctx context.Context, eksAPI sdk.EKSAPI, arczonalshiftAPI sdk.ARCZonalShiftAPI) (string, error) {
 	inputDC := eks.DescribeClusterInput{Name: &options.FromContext(ctx).ClusterName}
-	outputDC, _ := eksAPI.DescribeCluster(ctx, &inputDC)
+	outputDC, err := eksAPI.DescribeCluster(ctx, &inputDC)
+	if err != nil {
+		return "", fmt.Errorf("failed to describe cluster for zonal shift enablement, %w", err)
```
Comment (Ignoramuss) [inline kwok/operator/operator.go:360]: styling nit: `"failed to describe cluster for zonal shift enablement: %w"`

**Review by Sarthug99 — COMMENTED** [review]
We should add a unit test for this behavior. The mock APIs already exist, so should be straightforward?

File: pkg/operator/operator.go
Comment (Ignoramuss) [inline pkg/operator/operator.go:144]: Note: If DescribeCluster panics then clusterArn will be empty from the repsonse

**Review by ankit373 — COMMENTED** [review]
This fixes the panic in #9151 and is right as far as it goes. It has been sitting since May while the crash is still on `main`, so it would be good to land. Two things in the same four lines are still unguarded, and both produce the identical `SIGSEGV` the issue reports.

| Severity | Line | Finding |
|---|---|---|
| Medium | `clusterArn := outputDC.Cluster.Arn` | `err == nil` does not imply `outputDC.Cluster != nil` |
| Medium | `return *clusterArn, nil` | `Arn` is a `*string` and is dereferenced without a check |

On the first: the AWS SDK Go v2 gives no guarantee that a nil error means a fully populated struct, and `sdk.EKSAPI` is an interface, so any fake in tests or any future wrapper returning `&eks.DescribeClusterOutput{}` panics here exactly as before. The guard costs one line.

On the second: `GetManagedResource` is called with `ResourceIdentifier: clusterArn`. If that pointer is nil the call may well come back with a service-side error and take the early return, but that is the API deciding whether Karpenter crashes at startup. If it does not error, `*clusterArn` dereferences nil.

Both fold into one check:

```go
outputDC, err := eksAPI.DescribeCluster(ctx, &inputDC)
if err != nil {
    return "", fmt.Errorf("describing cluster for zonal shift enablement, %w", err)
}
if outputDC == nil || outputDC.Cluster == nil || outputDC.Cluster.Arn == nil {
    return "", fmt.Errorf("cluster %q has no ARN in the DescribeCluster response", options.FromContext(ctx).ClusterName)
}
clusterArn := outputDC.Cluster.Arn
```

after which `*clusterArn` at the return is safe.

One thing worth deciding separately from this PR. The caller does:

```go
panic(fmt.Sprintf("Unable to find Cluster %v in Zonal Shift. Please check that the cluster is enabled for Zonal Shift and roles have appropriate permissions %v", clusterArn, err))
```

On every error path this function returns `""`, so the message renders as `Unable to find Cluster  in Zonal Shift`, with a blank where the cluster name should be, and it asserts the cluster is not registered in Zonal Shift even when the real cause was `AccessDenied` on `eks:DescribeCluster`. That is the part that cost @jubrad time on #9151: the message sends you to check Zonal Shift registration when the answer is IAM. Returning the cluster name alongside the error, or having the caller print `options.FromContext(ctx).ClusterName` rather than the empty return value, fixes the misdirection. Happy to raise that separately if you would rather keep this PR to the panic.

The docs half of #9151 is already covered by #9171.

If you have moved on from this one, I am happy to pick it up with the extra guards and a table test over the three shapes (API error, nil `Cluster`, nil `Arn`). Just say so and I will open it against your branch or a fresh one.
